src/main/java/com/example/springbootrestexample/configurations/AppConfig.java

```diff
@@ -0,0 +1,46 @@
+package com.example.springbootrestexample.configurations;
+
+import com.example.springbootrestexample.repositories.*;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class AppConfig {
+
+    @Bean(name="bookRepository")
+    @Conditional(MongoDBCondition.class)
+    public BookRepository bookRepositoryInMongoDB() {
+        return new BookRepositoryInMongoDB();
+    }
+
+    @Bean(name="bookRepository")
+    @Conditional(MemoryConfiguration.class)
```

Comment:
Do not base the configurations in the @Conditional base those in the spring environment. It can be configurated in @Configuration and set in the program arguments:  -Dspring.profiles.active=
I would recommend to read: https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-profiles.html